### PR TITLE
feature (webapi): allow changing of user’s password by system admin

### LINKED ISSUE #795: Allow changing of user passwords by system administrator
needed for https://github.com/dhlab-basel/Salsah/issues/128

### PR DESCRIPTION
**Pull request by subotic**
### Summary

- system admin can now change passwords of other users
- removed code for checking SHA-1 hashed passwords. remnants of times gone by.

### Issues

- resolves #795

### CODE REVIEW

**Comment by subotic** [comment]
This was a very mean bug. I hate when a small problem takes almost the whole day to fix.

**Comment by subotic** [comment]
@benjamingeer I hope that the test will pass now, and then you can review whenever you have a spare moment. It's only a small PR.

File: webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala [hunk 1]
```diff
@@ -302,7 +302,7 @@ trait Authenticator {
             log.debug("getUserProfileV1 - I got a UserProfileV1: {}", user.toString)
 
             /* we return the userProfileV1 without sensitive information */
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala]: Is this comment still true?

Comment (subotic) [reply webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala]: nope, just changed it. thanks.

**Comment by benjamingeer** [comment]
Great, thanks!

**Comment by subotic** [comment]
thanks :-)

**Pull request merged**